src/anndata/compat/__init__.py

```diff
@@ -32,7 +34,23 @@ class Empty:
     pass
 
 
-Index1D = slice | int | str | np.int64 | np.ndarray | pd.Series
+Index1DNorm = slice | NDArray[np.bool] | NDArray[np.integer]
+# TODO: pd.Index[???]
+Index1D = (
+    # 0D index
+    int
+    | str
+    | np.int64
+    # normalized 1D idex
+    | Index1DNorm
+    # different containers for mask, obs/varnames, or numerical index
+    | pd.Series  # bool, int, str
+    | pd.Index
```

Comment:
I don’t think we test these (at least not via `subset_func` and therefore with `test_views_of_views`